### ISSUE TITLE
Regression: CORS doesn't work in 2.1.0

### ISSUE THREAD

**Issue by antonu17** [emissary-ingress/emissary#4006]
**Describe the bug**

A `Mapping` with CORS parameters renders a broken envoy configuration.

**To Reproduce**

1. Deploy emissary-ingress 2.1.0
2. Add an example `Mapping`:

```yaml
apiVersion: getambassador.io/v2
kind: Mapping
metadata:
  name: endpoint-with-cors
spec:
  cors:
    credentials: true
    origins: '*'
  host: localhost
  prefix: /endpoint-with-cors
  rewrite: /anything/cors
  service: http://httpbin.default
```

3. Optionally, restart the emissary-ingress pod

**Additional context**

The envoy fails with the following error in logs:
```
[2022-01-12 08:51:02.117][52][critical][main] [source/server/server.cc:109] error initializing configuration 'snapshots/econf-tmp.json': Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[4].route.cors) v2CommaSeparatedOrigins: Cannot find field.) has unknown fields
[2022-01-12 08:51:02.117][52][info][main] [source/server/server.cc:783] exiting
Protobuf message (type envoy.config.bootstrap.v3.Bootstrap reason INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[4].route.cors) v2CommaSeparatedOrigins: Cannot find field.) has unknown fields
```

Making `origins` a YAML array fixes the issue, but according to the [Mapping CORS documentation ](https://www.getambassador.io/docs/edge-stack/latest/topics/using/cors/#the-cors-attribute) (sorry if the link doesn't work, documentation structure is changing way too often, and sometimes links get broken, but that's a different issue), both syntaxes should work.

> Format can be either of comma-separated list or YAML array.

I noticed that a rogue entry gets rendered into envoy configuration file `econf.json`: `"v2CommaSeparatedOrigins": true`

```json
                            "route": {
                              ...
                              "cors": {
                                "allow_credentials": true,
                                "allow_origin_string_match": [
                                  {
                                    "exact": "*"
                                  }
                                ],
                                "filter_enabled": {
                                  "default_value": {
                                    "denominator": "HUNDRED",
                                    "numerator": 100
                                  },
                                  "runtime_key": "routing.cors_enabled.6f1913f011dc0ec183fcfdcf3aaded17c77899f3"
                                },
                                "v2CommaSeparatedOrigins": true
                              },
                              "prefix_rewrite": "/anything/cors",
                              ...
                            }
                          },
```

**Comment by khussey** [comment]
Thanks for the report, @antonu17. We're working on a fix and will be shipping a 2.1.1 release to address it ASAP.

**Comment by khussey** [comment]
This was fixed in Emissary-ingresss 2.1.2, which is now available.
